sdcm/sct_runner.py

```diff
@@ -930,39 +1036,107 @@ def list_sct_runners(test_status: Optional[str] = None, test_runner_ip: str = No
+        runner_to_update = [runner for runner in listed_runners if runner.test_id == test_id]
+
+    if not runner_to_update:
+        raise RuntimeError(f"Could not find SCT runner with IP: {test_runner_ip} to update tags for.")
+
+    try:
+        runner_to_update = runner_to_update[0]
+        runner_to_update.sct_runner_class.set_tags(runner_to_update, tags=tags)
+        LOGGER.info("Tags on SCT runner updated with: %s", tags)
+    except Exception as exc:  # pylint: disable=broad-except
+        LOGGER.warning("Could not set SCT runner tags to: %s due to exc:\n%s", tags, exc)
+
+
+def _manage_runner_keep_tag_value(utc_now: datetime,
+                                  timeout_flag: bool,
+                                  test_status: str,
+                                  sct_runner_info: SctRunnerInfo) -> SctRunnerInfo:
+    LOGGER.info("Managing runner's tags. Timeout flag: %s, logs_collected: %s",
+                timeout_flag, sct_runner_info.logs_collected)
+
+    if test_status == "SUCCESS" and sct_runner_info.logs_collected:
+        sct_runner_info.sct_runner_class.set_tags(sct_runner_info, {"keep": "0", "keep-action": "terminate"})
+        sct_runner_info.keep = 0
+        sct_runner_info.keep_action = "terminate"
+        return sct_runner_info
+
+    if not timeout_flag and sct_runner_info.logs_collected:
+        current_run_time_hrs = int((utc_now - sct_runner_info.launch_time).total_seconds() // 3600)
+        new_keep_value = int(sct_runner_info.keep) - current_run_time_hrs + 6
+
+        if new_keep_value > 0:
+            sct_runner_info.sct_runner_class.set_tags(sct_runner_info, {"keep": str(new_keep_value)})
+            sct_runner_info.keep = new_keep_value
+        return sct_runner_info
+
+    if not sct_runner_info.logs_collected:
+        sct_runner_info.sct_runner_class.set_tags(sct_runner_info, {"keep": "alive"})
+        sct_runner_info.keep = "alive"
+
+    return sct_runner_info
+
+
+def clean_sct_runners(test_status: str,
+                      test_runner_ip: str = None,
+                      dry_run: bool = False) -> None:
+    sct_runners_list = list_sct_runners(test_runner_ip=test_runner_ip)
+    timeout_flag = False
+    end_message = ""
+
+    if sct_runners_list:
+        sct_runner_info = sct_runners_list[0]
+        LOGGER.info("Attempting to create runner remoter with host: %s, region_name: %s",
+                    test_runner_ip, sct_runner_info.region_az)
+        runner_remoter = sct_runner_info.sct_runner_class(region_name=sct_runner_info.region_az,
+                                                          availability_zone="").get_remoter(host=test_runner_ip)
+        cmd = 'cat /home/ubuntu/sct-results/latest/events_log/critical.log | grep "TestTimeoutEvent"'
+        timeout_flag = bool(runner_remoter.run(cmd, ignore_status=True).stdout)
+        utc_now = datetime.datetime.now(tz=datetime.timezone.utc)
+        LOGGER.info("UTC now: %s", utc_now)
+
+        runners_cleaned = 0
```

Comment:
I think we don't need it more.

Reply:
Good catch. Fixed (removed, edited the final log message)